server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -1003,23 +1030,24 @@ protected void doXContentBody(XContentBuilder builder, Params params) throws IOE
         // this is a pain, but we have to do this to maintain BWC
         boolean includeDefaults = params.paramAsBoolean("include_defaults", false);
         builder.field("type", contentType());
-        this.builder.index.toXContent(builder, includeDefaults);
-        this.builder.store.toXContent(builder, includeDefaults);
+        final Builder b = (Builder) getMergeBuilder();
```

Comment:
Not the nicest approach to this I guess, but it should create the correct serialization as far as I understand?